File: 118.Pascal'sTriangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
    	vector<vector<int>> res;
    	res.resize(numRows);
    	for(int k = 0; k < numRows; k++){
    		res[k].resize(k+1);
    		for(int i = 0; i <= k; i++){
    			if(i == 0 || i == k) res[k][i] = 1;
    			else if(i > 0 && i < k){
    				if(k - 1 >= 0){
    					res[k][i] = res[k-1][i-1] + res[k-1][i];
    				}
    			}
    		}
    	}
    	return res;        
    }
};

File: 118.Pascal'sTriangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
    	List<List<Integer>> vecRes = new ArrayList<List<Integer>>();
    	for(int i = 0 ; i < numRows ; i++){
    		List<Integer> list = new ArrayList<Integer>();
    		for(int j = 0 ; j <= i ; j++){
    			if(j == 0 || j == i) list.add(1);
    			else 
    				list.add(vecRes.get(i-1).get(j-1)+vecRes.get(i-1).get(j));		                                          
    		}
    		vecRes.add(list);
    	}
        return vecRes;
    }
}                                                                                